src/rgw/rgw_sal_motr.cc

```diff
@@ -726,7 +726,7 @@ int MotrBucket::put_info(const DoutPrefixProvider *dpp, bool exclusive, ceph::re
 
   // Insert bucket instance using bucket's marker (string).
   int rc = store->do_idx_op_by_name(RGW_MOTR_BUCKET_INST_IDX_NAME,
-                                  M0_IC_PUT, tenant_bkt_name, bl, !exclusive);
```

Comment:
@selvakumaar5496 can you add comment here in review, stating how this change is making the difference

Reply:
Hi @cdeshmukh , @sachinpunadikar , @kupranay ,
The above change is incorrect.
```c++
int MotrStore::do_idx_op_by_name(string idx_name, enum m0_idx_opcode opcode,
                                 string key_str, bufferlist &bl, bool update)
```
the last argument of `do_idx_by_name()` function is `update` which is the opposite of `exclusive`.
Meaning `not` of `exclusive` should be passed to `update` arg.

this patch is causing `BucketAlreadyExists` issues in APIs which update the bucket metadata, for example, `put-bucket-quota` or `set-bucket-versioning` etc. 
These APIs are correctly passing `exclusive = false` arg to `MotrBucket::put_info()`, but the wrong interpretation of exclusive arg is causing the error.
```bash
Wed, 4 May 2022 08:50:10 GMT
/admin/bucket
PUT http://ssc-vm-g4-rhev4-0318.colo.seagate.com:8000/admin/bucket?quota&bucket=mybkt&uid=mgwuser&max-size-kb=1024&enabled=true HTTP/1.1
Auth: AWS NB71GXG3O7MMBELODQOO:m4CTpp1L4xQ3IYmdd0p3Vuw+UI0=
HTTP/1.1 409 Conflict

Response Content is: {"Code":"BucketAlreadyExists","HostId":""}
```

Reply:
For `PutBucketTagging`, the following sal layer implementation is wrong.
```c++
int MotrBucket::merge_and_store_attrs(const DoutPrefixProvider *dpp, Attrs& new_attrs, optional_yield y)
{
  for (auto& it : new_attrs)
    attrs[it.first] = it.second;

  return put_info(dpp, y, ceph::real_time());
}
```
here the `put_info()` should pass `exclusive = false` as an argument.